### PR TITLE
[gulp] new task 'upload-binaries' (to github)

### PR DESCRIPTION
**Pull request by luclu**
This PR allows Travis to upload the generated binaries (`dist-${type}/release`) to github.

It will only do so if:
1. building on `master` branch (after a PR got merged from `develop` or direct commit)
2. a git tag `v{version}` exists
3. a release in **draft** mode coupled with the tag from (2) exists
4. the draft does not contain any files with the same name that the generated binaries have

This solves:
- direct CI upload
- can't use electron-builder's internal uploader because:
  - naming problem using electron-builder to upload (#1570) 
  - using custom NSIS script (https://github.com/ethereum/mist/pull/1499) to create windows installer a posteriori 

Also it will allow for simple integration of upcoming azure uploader.

I extensively tested this PR on my fork of the repo.

### CODE REVIEW

**Comment by frozeman** [comment]
So the tag needs to exist before while the release is still in draft?
I would suggest that the release creates the tag on publish how we do it now

**Comment by luclu** [comment]
Eventually we decided to create the tag a prior - Ev reminded me that to sign tags those must be created locally pushed. Did I understand this correct, @evertonfraga?

Otherwise letting Travis create the tag + draft if the version gets bumped on `master` sounds resonable. (Although the upload should also work if the draft already exists/was created by us beforehand.)

**Comment by alexvandesande** [comment]
Thanks for this @luclu please review the conflicts

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ cache:
 branches:
   only:
     - develop
```
Comment (frozeman) [inline .travis.yml]: Why we upload on develop?

File: gulpfile.js [hunk 2]
```diff
@@ -347,14 +356,54 @@ gulp.task('release-dist', ['build-dist'], (done) => {
     done();
 });
 
+gulp.task('upload-binaries', () => {
+    // token must be set using travis' ENVs
+    const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
+
+    // query github releases
+    return got(`https://api.github.com/repos/ethereum/mist/releases?access_token=${GITHUB_TOKEN}`, {
```
Comment (frozeman) [inline gulpfile.js]: Is that token secret?

Comment (frozeman) [inline gulpfile.js]: Ah i see it comes from travis

File: .travis.yml (hunk 1 above)
Comment (luclu) [inline .travis.yml]: The `branch -> only` array includes all branches that should trigger a travis build. Thus we want `develop` and now `master` for the upload.

File: gulpfile.js (hunk 2 above)
Comment (luclu) [inline gulpfile.js]: Correct.

**Pull request merged**